backend/src/main/java/net/es/oscars/web/rest/VirtInterfaceController.java

```diff
@@ -0,0 +1,283 @@
+                }
+            }
+        }
+        // if virt ip doesn't exist -> add virt ip
+        NsoVirtInterface instanceToModify = null;
+        if (!ipExists) {
+            instanceToModify = cache.get(device);
+
+            // if no entry for this devices exists -> create new entry
+            if (instanceToModify == null) {
+                instanceToModify = new NsoVirtInterface();
+                instanceToModify.setDevice(device);
+                instanceToModify.setConnectionId(connectionId);
+            }
+
+            // add virt ip
+            instanceToModify.getIpAddresses().add(ipAndSubnet);
+
+            // add to DB
+            virtInterfaceDAO.save(instanceToModify);
+
+            log.info("Virt IP " + ipAndSubnet + " added to " + device + " for connection id " + connectionId);
+        } else {
+            log.info("Virt IP " + ipAndSubnet + " already exists");
+            throw new IllegalArgumentException();
+        }
+        return;
+    }
+
+
+    @RequestMapping(value = "/api/interface/remove", method = RequestMethod.POST)
+    @ResponseBody
+    @Transactional
+    public void removeVirtIpInterface(@RequestBody VirtIpInterfaceRequest request) {
+        log.debug("request: " + request.toString());
+
+        verifyData(request);
+
+        String connectionId = request.getConnectionId();
+        String device = request.getDevice();
+        String ipAndSubnet = request.getIpAndSubnet();
+
+        List<NsoVirtInterface> interfaces = virtInterfaceDAO.findByConnectionId(connectionId);
+        if(interfaces == null || interfaces.isEmpty()) {
+            log.info("No virt IPs found");
+            throw new NoSuchElementException();
+        }
+
+        boolean virtIpFound = false;
+        for (NsoVirtInterface ifce : interfaces) {
+            for (String ipAddress : ifce.getIpAddresses()) {
+                if(ipAndSubnet.equals(ipAddress)) {
+                    virtIpFound = true;
+
+                    // if last ip address remove entry form DB else modify IP address list
+                    if (ifce.getIpAddresses().size() <= 1) {
+                        virtInterfaceDAO.deleteById(ifce.getId());
+                    } else {
+                        ifce.getIpAddresses().remove(ipAddress);
+                        virtInterfaceDAO.save(ifce);
+                    }
+
+                    log.info("Virt IP " + ipAndSubnet + " removed from " + device + " with connection id " + connectionId);
+                    break;
+                }
+            }
+        }
+
+        if(!virtIpFound) {
+            log.info("No virt IP " + ipAndSubnet + " found for connection id " + connectionId);
+            throw new NoSuchElementException();
+        }
+    }
+
+
+    @RequestMapping(value = "/api/interface/list/{connectionId}", method = RequestMethod.GET)
+    @ResponseBody
+    @Transactional
+    public VirtIpInterfaceListResponse listVirtIpInterface(@PathVariable String connectionId) {
```

Comment:
I haven't tried but AFAIK we could return just the list and don't need to warp this into an own Response object ... 
currently it looks like this:
`{
    "output": [
        {
            "device": "loc1-cr6",
            "ip-address": [
                "192.168.1.1/24",
                "192.168.1.2/24"
            ]
        },
        {
            "device": "loc2-cr6",
            "ip-address": [
                "192.168.1.11/24",
                "192.168.1.12/24"
            ]
        }
    ]
}`

Reply:
I agree you should just return the List<>

Reply:
okay, let me get rid of the object class and change the return type to a list